__init__.py

```diff
@@ -61,93 +63,280 @@ def mall_link(self):
-            if shop_by_floor == 'yes':
-                floor = self.get_response('which_floor')
-                shop = self.request_handler.shop_selection_by_floors(floor, shop_info)
-                if shop is not None:
-                    return self.speak_shops([shop])
+            self.speak_dialog('no_shop_on_level')
+            self.speak_shops(shop_info)
+        return 3, None
+
+    def open_shops_search(self, shop_info, day_time, hour, min):
+        """
+       Selects open shops. Collects the list of
+       open shops else return empty list.
+       Args:
+           shop_info (list): found shops on user's
+                               request
+       Returns:
+           shop_info (list): open shops
+       """
+        open_shops = []
+        LOG.info(f"User's time {day_time, hour, min}")
+        for shop in shop_info:
+            parse_time = re.findall(r'(\d+)+[am|pm]', shop['hours'])
+            LOG.info(f'Parsed time {parse_time}')
+            open_time = int(parse_time[0])
+            close_time = int(parse_time[1])
+            if open_time <= hour < close_time:
+                open_shops.append(shop)
+            elif day_time[1] == 'am' and open_time <= hour:
+                open_shops.append(shop)
+        return open_shops
+
+
+    def time_calculation(self, shop_info, open, day_time, hour, min):
+        # add logic if shop opens and closes not at am-pm time period
+        # change to speak dialog
+        """
+        Calculates time difference between user's current time
+        and shop working hours.
+        If 'open' argument is True:
+            If user one hour or less before closing: speaks how
+                many minutes left. Speaks shop info.
+            Else speaks corresponging dialog. 
+            Speaks shop info.
+        If 'open' argument is False:
+            Speaks corresponding dialog.
+            If user is one hour or less before opening hours 
+                speaks how much time is left for waiting. 
+            If user's time is 'am' and user is before opening
+                hours, speaks how many hours and minutes left 
+                waiting.
+            If user's time is evening (pm) speaks when the shop 
+                opens in the morning.
+                Speaks shop info.
+        Args:
+            shop_info (list): found shops on user's request
+            open (boolean): True - if shop is open
+            day_time (str): user's current day time (am|pm) 
+            hour (int): user's current hour
+            min (int): user's current minute
+        Returns:
+            3, None (to ask for another shop info)
+        Examples:
+            work time 9am-10pm
+            user's time 8am
+            Prompt: 'Shop is closed now. Opens in 1 hour'
+        """
+        for shop in shop_info:
+            work_time = shop['hours']
+            normalized_time = re.findall(r'(\d+)[am|pm]', work_time)
```

Comment:
Just a note that using `datetime` might simplify things in the future; the parsing here looks good though